packages/expression-evaluator/lib/evaluators/ExpressionEvaluatorFactory.ts

```diff
@@ -1,42 +1,73 @@
-import type { IAsyncEvaluatorContext } from './ExpressionEvaluator';
+import type * as E from '../expressions';
+import type { TermSparqlFunction, SparqlFunction } from '../functions';
+import { namedFunctions, regularFunctions, specialFunctions } from '../functions';
+import type * as C from '../util/Consts';
+import { RegularOperator } from '../util/Consts';
+import type { IAsyncEvaluatorContext } from './ContextualizedEvaluator';
+import { ContextualizedEvaluator } from './ContextualizedEvaluator';
 import { ExpressionEvaluator } from './ExpressionEvaluator';
-
-/**
- * A counter that keeps track blank node generated through BNODE() SPARQL
- * expressions.
- *
- * @type {number}
- */
-let bnodeCounter = 0;
+import { OrderByEvaluator } from './OrderByEvaluator';
 
 export class ExpressionEvaluatorFactory implements IExpressionEvaluatorFactory {
   public readonly mediatorBindingsAggregatorFactory: MediatorBindingsAggregatorFactory;
   public readonly mediatorQueryOperation: MediatorQueryOperation;
+  public readonly functionsBus: FunctionBusType = async({ functionName }) => {
+    const res: SparqlFunction | undefined = {
+      ...regularFunctions,
+      ...specialFunctions,
+      ...namedFunctions,
+    }[<C.NamedOperator | C.Operator> functionName];
+    if (res) {
+      return res;
+    }
+    throw new Error('nah!');
+  };
+
+  public readonly termFunctionsBus: TermFunctionBusType = async({ functionName }) => {
+    const res: TermSparqlFunction<any> | undefined = {
+      ...regularFunctions,
+      ...namedFunctions,
+    }[<C.NamedOperator | C.RegularOperator> functionName];
+    if (res) {
+      return res;
+    }
+    throw new Error('nah!');
+  };
 
   public constructor(args: IExpressionEvaluatorFactoryArgs) {
     this.mediatorBindingsAggregatorFactory = args.mediatorBindingsAggregatorFactory;
     this.mediatorQueryOperation = args.mediatorQueryOperation;
   }
 
   // TODO: remove legacyContext in *final* update (probably when preparing the EE for function bussification)
-  public createEvaluator(algExpr: Alg.Expression, context: IActionContext,
-    legacyContext: Partial<IAsyncEvaluatorContext> = {}): ExpressionEvaluator {
-    return new ExpressionEvaluator(algExpr, {
-      now: context.get(KeysInitQuery.queryTimestamp),
-      baseIRI: context.get(KeysInitQuery.baseIRI),
-      functionArgumentsCache: context.get(KeysInitQuery.functionArgumentsCache),
-      actionContext: context,
-      bnode: (input?: string) => Promise.resolve(new BlankNodeBindingsScoped(input || `BNODE_${bnodeCounter++}`)),
-      exists: ActorQueryOperation.createExistenceResolver(context, this.mediatorQueryOperation),
-      ...legacyContext,
-    });
+  public async createEvaluator(algExpr: Alg.Expression, context: IActionContext,
+    contextEval?: ContextualizedEvaluator,
+    legacyContext: Partial<IAsyncEvaluatorContext> = {}): Promise<IExpressionEvaluator> {
+    if (!contextEval) {
+      const defContextEval = new ContextualizedEvaluator({
```

Comment:
I guess this wouldn't be needed anymore if `ContextualizedEvaluator` gets merged into `ExpressionEvaluator`.